src/main/kotlin/de/itemis/mps/gradle/modelcheck/Plugin.kt

```diff
@@ -64,36 +63,17 @@ open class ModelcheckMpsProjectPlugin : Plugin<Project> {
                     args.add("--result-format=${extension.junitFormat}")
                 }
 
-                val resolveMps = tasks.create("resolveMpsForModelcheck", Copy::class.java) {
-                    from(extension.mpsConfig.resolve().map { zipTree(it) })
-                    /*
-                    Some 2019.2 src jars cause MPS startup to fail with an exception during indexing
-                    because of the wrong specification in module.xml -> we remove those broken parts as a workaround.
-                    In MPS 2019.3 those src jars are already removed.
-                     */
-                    val modelsTagRegEx = Regex("<models>.*</models>", RegexOption.DOT_MATCHES_ALL)
-                    val replaceTag = "<models/>"
-                    filesMatching(listOf("**/runtimes/*.feedback.*-src.jar", "**/runtimes/*.messages.api-src.jar")) {
-                        ZipInputStream(open()).use { zis ->
-                            val resultEntries = ArrayList<Pair<String, ByteArray>>()
-                            do {
-                                val nextEntry = zis.nextEntry ?: break
-                                val outputStream = ByteArrayOutputStream()
-                                zis.copyTo(outputStream)
-                                var byteArray : ByteArray
-                                if (nextEntry.name.contains(".msd")) {
-                                    byteArray = outputStream.toString().replace(modelsTagRegEx, replaceTag).toByteArray()
-                                } else {
-                                    byteArray = outputStream.toByteArray()
-                                }
-                                resultEntries.add(Pair(nextEntry.name, byteArray))
-                            } while (true)
-                            zipTo(File(mpsLocation, this.path), resultEntries.asSequence())
-                        }
-                        exclude()
+
+                val resolveMps: Task = if (extension.mpsConfig != null) {
+                    tasks.create("resolveMpsForModelcheck", Copy::class.java) {
+                        from(extension.mpsConfig!!.resolve().map { zipTree(it) })
```

Comment:
This downloads MPS at the configuration time :(
In other words, it would download MPS even for `./gradlew help`.

I would suggest something like the following

```kotlin
dependsOn(mpsConfig)
from(zipTree(provider { mpsConfig.singleFile }))
```